simulation/traffic_simulator/test/src/test_hdmap_utils.cpp

```diff
@@ -122,6 +122,116 @@ TEST(HdMapUtils, RoadShoulder)
+  EXPECT_EQ(
+    canonicalized_lanelet_pose.value().s, canonicalized_lanelet_offset +
+                                            hdmap_utils.getLaneletLength(34570) +
+                                            hdmap_utils.getLaneletLength(34576));
+}
+
+TEST(HdMapUtils, CanonicalizePositive)
+{
+  std::string path =
+    ament_index_cpp::get_package_share_directory("traffic_simulator") + "/map/lanelet2_map.osm";
+  geographic_msgs::msg::GeoPoint origin;
+  origin.latitude = 35.61836750154;
+  origin.longitude = 139.78066608243;
+  hdmap_utils::HdMapUtils hdmap_utils(path, origin);
+
+  double non_canonicalized_lanelet_s = 30;
+  const auto non_canonicalized_lanelet_pose =
+    traffic_simulator::helper::constructLaneletPose(34981, non_canonicalized_lanelet_s, 0);
+  const auto canonicalized_lanelet_pose = std::get<std::optional<traffic_simulator::LaneletPose>>(
+    hdmap_utils.canonicalizeLaneletPose(non_canonicalized_lanelet_pose));
+
+  EXPECT_EQ(canonicalized_lanelet_pose.value().lanelet_id, 34579);
+  EXPECT_EQ(
+    canonicalized_lanelet_pose.value().s, non_canonicalized_lanelet_s -
+                                            hdmap_utils.getLaneletLength(34981) -
+                                            hdmap_utils.getLaneletLength(34585));
+}
+
+TEST(HdMapUtils, CanonicalizeAllNegative)
+{
+  std::string path =
+    ament_index_cpp::get_package_share_directory("traffic_simulator") + "/map/lanelet2_map.osm";
+  geographic_msgs::msg::GeoPoint origin;
+  origin.latitude = 35.61836750154;
+  origin.longitude = 139.78066608243;
+  hdmap_utils::HdMapUtils hdmap_utils(path, origin);
+
+  double canonicalized_lanelet_offset = -22;
+  const auto non_canonicalized_lanelet_pose =
+    traffic_simulator::helper::constructLaneletPose(34564, canonicalized_lanelet_offset, 0);
+  const auto canonicalized_lanelet_poses =
+    hdmap_utils.gelAllCanonicalizedLaneletPoses(non_canonicalized_lanelet_pose);
+
+  EXPECT_EQ(canonicalized_lanelet_poses.size(), static_cast<long unsigned int>(3));
+  EXPECT_EQ(canonicalized_lanelet_poses[0].lanelet_id, 34576);
+  EXPECT_EQ(
+    canonicalized_lanelet_poses[0].s, canonicalized_lanelet_offset +
+                                        hdmap_utils.getLaneletLength(34570) +
+                                        hdmap_utils.getLaneletLength(34576));
+  EXPECT_EQ(canonicalized_lanelet_poses[1].lanelet_id, 34981);
+  EXPECT_EQ(
+    canonicalized_lanelet_poses[1].s, canonicalized_lanelet_offset +
+                                        hdmap_utils.getLaneletLength(34636) +
+                                        hdmap_utils.getLaneletLength(34981));
+  EXPECT_EQ(canonicalized_lanelet_poses[2].lanelet_id, 34600);
+  EXPECT_EQ(
+    canonicalized_lanelet_poses[2].s, canonicalized_lanelet_offset +
+                                        hdmap_utils.getLaneletLength(34648) +
+                                        hdmap_utils.getLaneletLength(34600));
+}
+
+TEST(HdMapUtils, CanonicalizeAllPositive)
+{
```

Comment:
Please describe the test target and expected result in comment.

Reply:
Comment added

Reply:
I've changed the comment format -> 177a344d7c55afa1d27d8946380ff00a468b40bd